src/Subscriptions/Subscriber.php

```diff
@@ -24,6 +24,12 @@ class Subscriber
      */
     public $channel;
 
+    /**
+     * X-Socket-ID header passed on the subscription query.
+     * @var string|null|array<mixed>
```

Comment:
I think we should validate this is either a string or null

Reply:
Agreed array doesn't really make much sense here yet again if I add this:

```.php
        $XSocketID = request()->header('X-Socket-ID');
        if (! is_array($XSocketID)) {
            $this->socket_id = $XSocketID;
        }
```
I gives an error on my php version (8.1) Laravel (9)

```
 100    Call to function is_array() with string|null will always evaluate to false.
```

Kind of stuck here...

Reply:
And even if I use this:

```php
        $XSocketID = request()->header('X-Socket-ID');
        if (is_string($XSocketID) || is_null($XSocketID)) {
            $this->socket_id = $XSocketID;
        }
```

Php stan gives:

```
  100    Result of || is always true.      
```

Reply:
I would throw if the value is an array. Just try and push it to see if the different versions disagree, then maybe add a `@phpstan-ignore-next-line`. I am fixing the dependencies in https://github.com/nuwave/lighthouse/pull/2300.